src/gridlayout.jl

```diff
@@ -1,45 +1,48 @@
-GridLayout(; kwargs...) = GridLayout(1, 1; kwargs...)
+GridLayout(; @nospecialize(kwargs...)) = GridLayout(1, 1; kwargs...)
 
 observablify(x::Observable) = x
-observablify(x, type=Any) = Observable{type}(x)
+observablify(x) = Observable(x)
+struct Observablify{T} end
+Observablify{T}(x) where T = convert(Observable{T}, x)::Observable{T}
 
 function GridLayout(nrows::Int, ncols::Int;
         parent = nothing,
         rowsizes = nothing,
         colsizes = nothing,
         addedrowgaps = nothing,
         addedcolgaps = nothing,
-        alignmode = Inside(),
+        alignmode::AlignMode = Inside(),
         equalprotrusiongaps = (false, false),
         bbox = nothing,
         width::SizeAttribute = Auto(),
         height::SizeAttribute = Auto(),
         tellwidth::Bool = true,
         tellheight::Bool = true,
-        halign = :center,
-        valign = :center,
-        default_rowgap = DEFAULT_ROWGAP_GETTER[](),
-        default_colgap = DEFAULT_COLGAP_GETTER[](),
+        halign::AlignAttribute = :center,
```

Comment:
In principle, an align is just a float. The symbols are just because this is directly user facing, and it's nicer to write `:right` rather than 1. But I already feared that these type instabilities that propagate relatively far into the code aren't good.